tf_quant_finance/volatility/bachelier_tf.py

```
@@ -0,0 +1,248 @@
+
+'Copyright 2020 Joerg Kienitz
+
+'Redistribution and use in source and binary forms, with or without modification,
+'are permitted provided that the following conditions are met:
+
+'1. Redistributions of source code must retain the above copyright notice,
+'this list of conditions and the following disclaimer.
+
+'2. Redistributions in binary form must reproduce the above copyright notice,
+'this list of conditions and the following disclaimer in the documentation
+'and/or other materials provided with the distribution.
+
+'3. Neither the name of the copyright holder nor the names of its contributors
+'may be used to endorse or promote products derived from this software without
+'specific prior written permission.
+
+'THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+'"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+'THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+'ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+'FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+'(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+'LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+'ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+'OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+'THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+@author: Joerg Kienitz
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+# straight fwd implementation of the Bachelier pricing
+# there is a version with just one call to exp !!
+
+    
+def option_price(spots, 
+                 strikes, 
+                 volatilities, 
+                 expiries,
+                 rates,
```

we are adding `rates` to black-scholes option price formula. Let's keep the names consistent.
Please call it `discount_rates` and also add `discount_factors`. Please raise an error if both supplied at the same time.